### DISCUSSION TITLE
How to Automatically Download the Latest Windows Device Drivers

### DISCUSSION THREAD

**Discussion by giscus** [TsekNet/website#16] [Announcements]
# How to Automatically Download the Latest Windows Device Drivers

Automatically download the latest device drivers during Windows OS deployment (Dell, Lenovo, HP, etc.).

https://tseknet.com/blog/modeldrivers

**Comment by Turmaxx** [comment]
Though I might not use Windows as much. I really liked this blog post. :thumbsup:.
Looking forward to more posts.

**Comment by kevins1501** [comment]
Hi! Just came across this and wanted to know how I can leverage this via MDT? Please advise!

**Reply by TsekNet** [reply]
Hey @kevins1501, thanks for the interest!

The script was designed with MDT (and similar deployment tools) in mind. Here's how to integrate it:

1. **Copy the script** into your MDT deployment share (e.g. `Scripts\Install-ModelDrivers\install_model_drivers.ps1`)

2. **Add a "Run PowerShell Script" task** in your MDT task sequence, after the OS is applied but before the first reboot. Point it to the script with the `-Manufacturer` and `-Model` parameters for your hardware:

```powershell
.\install_model_drivers.ps1 -Manufacturer Dell -Model 9380
```

3. **To auto-detect the model**, you can wrap the call with WMI so you don't need a separate task per model:

```powershell
$manufacturer = (Get-CimInstance Win32_ComputerSystem).Manufacturer
$model = (Get-CimInstance Win32_ComputerSystem).Model
.\install_model_drivers.ps1 -Manufacturer $manufacturer -Model $model
```

> **Note:** the `-Model` parameter is a regex match against the manufacturer's driver catalog, so you may need to use a partial string (e.g. `20y0` for a Lenovo ThinkPad) rather than the full WMI model name. Test the match for your specific hardware first.

4. **Network access is required** during the task sequence since the script downloads drivers directly from the manufacturer's website (Dell, Lenovo, HP, Microsoft).

The full script and examples are here: https://github.com/TsekNet/PowerShell/tree/master/Install-ModelDrivers
